### DISCUSSION TITLE
How to run Sygus in interactive mode?

### DISCUSSION THREAD

**Discussion by tmoux** [cvc5/cvc5#10111] [Q&A] [answered]
I'd like to run sygus programs in interactive mode.
For instance, when I save [this file](https://github.com/cvc5/cvc5/blob/main/examples/api/smtlib/sygus-fun.sy) to `sygus-fun.sy` and run `cvc5 --lang=sygus2 sygus-fun.sy`, it works as expected.
However, when I run `cvc5 --interactive --lang=sygus2`, I get the error `(error "internal error: unhandled language LANG_SYGUS_V2")`.

Looking at this, https://github.com/cvc5/cvc5/blob/eeeec1c75595b55668d271a9d77308b2aa683829/src/main/interactive_shell.cpp#L99-L106
it seems input languages other than smtlib aren't allowed in the interactive mode. Is there a reason for this? Is there any way around this to use sygus in interactive mode?

**Comment by ajreynol** [comment, marked as answer]
As far as I can tell, I don't think there is a reason to exclude sygus here. I can update this.
